tools/ci_build/github/apple/objectivec/objc.podspec.template

```diff
@@ -0,0 +1,54 @@
+Pod::Spec.new do |s|
+  s.name                  = "@NAME@"
+  s.version               = "@VERSION@"
+  s.summary               = "@SUMMARY@"
+  s.description           = "@DESCRIPTION@"
+
+  s.homepage              = "https://github.com/microsoft/onnxruntime-genai"
+  s.license	              = { :type => "MIT", :file => "@LICENSE_FILE@" }
+  s.author                = { "ONNX Runtime" => "onnxruntime@microsoft.com" }
+  s.source                = { :http => 'file:///http_source_placeholder' }
+  s.ios.deployment_target = "@IOS_DEPLOYMENT_TARGET@"
+  s.osx.deployment_target = "@MACOSX_DEPLOYMENT_TARGET@"
+
+  s.default_subspec     = 'Core'
+  s.static_framework    = true
+
+  s.subspec 'Core' do |core|
+    core.dependency "@C_POD_NAME@", "#{s.version}"
+    core.requires_arc   = true
+    core.compiler_flags = "-std=c++17", "-fobjc-arc-exceptions", "-Wall", "-Wextra", "-Werror"
+
+    include_dirs = [
+      @INCLUDE_DIR_LIST@
+    ].map { |relative_include_dir|
+      '"${PODS_TARGET_SRCROOT}/' + relative_include_dir + '"'
+    }
+
+    core.public_header_files = [
+      @PUBLIC_HEADER_FILE_LIST@
+    ]
+
+    core.source_files = [
+      @SOURCE_FILE_LIST@
+    ]
+
+    core.test_spec "Tests" do |test|
+      test.source_files = [
+        @TEST_SOURCE_FILE_LIST@
+      ]
+
+      test.resources = [
+        @TEST_RESOURCE_FILE_LIST@
+      ]
+    end
+
+    core.pod_target_xcconfig = {
+      "HEADER_SEARCH_PATHS" => include_dirs.join(" "),
+      "OTHER_CPLUSPLUSFLAGS" => "-fvisibility=hidden -fvisibility-inlines-hidden",
+    }
+
+  end
+  s.dependency 'onnxruntime-objc', '~> @ORT_VERSION@'
```

Comment:
why does it need to depend on onnxruntime-objc?

Reply:
i'm not sure about this. technically it doesn't *need* `onnxruntime-objc` at this moment. but I'm not sure if depending on `onnxruntime-c` will always suffice in the future. There isn't really harm including `onnxruntime-objc`, is there?

Reply:
> There isn't really harm including onnxruntime-objc, is there?

well, it's an additional thing to download. if we need the dependency later, we can add it then.

Reply:
removed